### PR TITLE
in non-strict mode return only if non-empty

### PR DESCRIPTION
**Pull request by sravfeyn**
http://manage.dimagi.com/default.asp?182228#edit_0_182228

### CODE REVIEW

**Comment by sravfeyn** [comment]
@esoergel

File: corehq/apps/app_manager/models.py [hunk 1]
```diff
@@ -1124,9 +1124,9 @@ def _get_media_by_language(self, media_attr, lang, strict=False):
         media_dict = getattr(self, media_attr)
         if not media_dict:
             return None
-        if lang in media_dict:
+        if media_dict.get(lang, ''):
             return media_dict[lang]
```
Comment (kaapstorm) [inline corehq/apps/app_manager/models.py]: That's going to raise a KeyError if lang is not in media_dict. Don't you mean

    if strict:
        return media_dict.get(lang, '')
    else:
        # ...

Comment (sravfeyn) [inline corehq/apps/app_manager/models.py]:
```python
>>> x = {}
>>> x.get('d',1)
1

Comment (kaapstorm) [inline corehq/apps/app_manager/models.py]: Sorry. Should have read the commit message. This should be:

    if lang in media_dict and media_dict[lang]:
        return media_dict[lang]

Comment (sravfeyn) [inline corehq/apps/app_manager/models.py]: Sorry

Comment (kaapstorm) [inline corehq/apps/app_manager/models.py]: Crap. I'm being an idiot. this is fine, because '' is falsey.

**Comment by kaapstorm** [comment]
:+1: just waiting for tests

File: corehq/apps/app_manager/models.py (hunk 1 above)
Comment (sravfeyn) [inline corehq/apps/app_manager/models.py]: It won't reach that line because `if media_dict.get(lang, '')` will evaluate to `False`

**Pull request merged**